src/app/admin/product-form/product-form.component.html

```diff
@@ -0,0 +1,95 @@
+<div class="row">
+  <div class="col-md-6">
+    <form [formGroup]="productForm" class="form-wrapper" (ngSubmit)="save()">
+      <div class="form-group">
+        <label for="title">Title</label
+        ><input
+          formControlName="title"
+          id="title"
+          type="text"
+          class="form-control"
+        />
+        <div class="alert alert-danger" *ngIf="title.touched && title.invalid">
+          <div *ngIf="title.errors.required">
+            Title is required
+          </div>
+        </div>
+      </div>
+      <div class="form-group">
+        <label for="price">Price</label>
+        <div class="input-group">
+          <span class="input-group-text">$</span>
+          <input
+            formControlName="price"
+            id="price"
+            type="number"
+            class="form-control"
+          />
+        </div>
+        <div class="alert alert-danger" *ngIf="price.touched && price.invalid">
+          <div *ngIf="price.errors.required">
+            Price is required.
+          </div>
+          <div *ngIf="price.errors.min">
+            Price should be 0 or higher.
+          </div>
+        </div>
+      </div>
+      <div class="form-group">
+        <label for="category">Category</label
+        ><select formControlName="category" id="category" class="form-control">
+          <option value=""></option>
+          <option *ngFor="let c of categories$ | async" [value]="c.key">
+            {{ c.payload.val().name }}
+          </option>
+        </select>
+        <div
+          class="alert alert-danger"
+          *ngIf="category.touched && category.invalid"
+        >
+          <div *ngIf="category.errors.required">
+            Category is required.
+          </div>
+        </div>
+      </div>
+      <div class="form-group">
+        <label for="imageUrl">Image URL</label
+        ><input
+          formControlName="imageUrl"
+          id="imageUrl"
+          type="url"
+          class="form-control"
+          pattern="https://.*"
+        />
+        <div
+          class="alert alert-danger"
+          *ngIf="imageUrl.touched && imageUrl.invalid"
+        >
+          <div *ngIf="imageUrl.errors.required">
+            Image URL is required.
```

Comment:
this should live in a const or an enum along with all the other validation errors